core/src/main/java/ai/timefold/solver/core/impl/localsearch/DefaultLocalSearchPhase.java

```diff
@@ -125,6 +125,7 @@ public void phaseStarted(LocalSearchPhaseScope<Solution_> phaseScope) {
         super.phaseStarted(phaseScope);
         decider.phaseStarted(phaseScope);
         assertWorkingSolutionInitialized(phaseScope);
+        phaseScope.setTermination(phaseTermination);
```

Comment:
This is on `AbstractPhaseScope`, yet only `LocalSearchPhase` actually sets a value.
This is a recipe for an NPE - either make it only available on `LocalSearchPhaseScope`, or make sure the termination is set for every phase type.